presentationLayer/ViewModels/PostcodeCheckerViewModel.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  PostcodeCheckerViewModel.swift
+//  fuelPrices
+//
+//  Created by Matthew Paul Harding on 26/04/2024.
+//
+
+import Foundation
+
+class PostcodeCheckerViewModel: ObservableObject {
+    
+    // properties for the UI View to "bind" to
+    @Published var filteredStations: [PetrolStation] = []
+    var postcode: String = ""
+    var isRefreshing: Bool { // the ViewModel is the only API (interface) that should be visible to the View, so we wrap this value. i.e. we never want to write code where we are chaining down through the layers of architecture - ❌ viewModel.stationDataManager.isRefreshingData // ✅ viewModel.isRefreshingData
+        stationDataManager.isRefreshingData
+    }
+    
+    // access to "the system" (i.e. business logic layer)
+    private var stationDataManager = StationDataManager(fuelProviders: FuelProviderDictionary.fuelProviders)
```

Comment:
The view model is aware of stationDataManager but the View doesn't know it exists. This is important. The view should only ever be aware of the viewmodel. The View should look very simple after this change and would be very easy to change for macOS, watchOS, GogglesOS etc.